src/bp-members/classes/class-bp-members-notices-rest-controller.php

```diff
@@ -0,0 +1,1090 @@
+						'readonly'    => true,
+						'type'        => 'integer',
+					),
+					'action_url'  => array(
+						'context'     => array( 'view', 'edit', 'embed' ),
+						'description' => __( 'The Notice Action URL.', 'buddypress' ),
+						'format'      => 'uri',
+						'type'        => 'string',
+					),
+					'action_text' => array(
+						'context'     => array( 'view', 'edit', 'embed' ),
+						'description' => __( 'The Notice Action text.', 'buddypress' ),
+						'type'        => 'string',
+					),
+				),
+			);
+		}
+
+		/**
+		 * Filters the notice schema.
+		 *
+		 * @since 15.0.0
+		 *
+		 * @param array $schema The endpoint schema.
+		 */
+		return apply_filters( 'bp_members_notices_rest_schema', $this->add_additional_fields_schema( $this->schema ) );
+	}
+
+	/**
+	 * Get the query params for notices collection.
+	 *
+	 * @since 15.0.0
+	 *
+	 * @return array
+	 */
+	public function get_collection_params() {
+		$params                       = parent::get_collection_params();
+		$params['context']['default'] = 'view';
+
+		if ( isset( $params['per_page']['default'] ) ) {
+			$params['per_page']['default'] = 5;
+		}
+
+		$params['user_id'] = array(
+			'description'       => __( 'Limit result set to items concerning a specific user (ID).', 'buddypress' ),
+			'default'           => 0,
+			'type'              => 'integer',
+			'sanitize_callback' => 'absint',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		$params['status'] = array(
+			'description'       => __( 'Limit result set to items with a specific status.', 'buddypress' ),
+			'default'           => 'unread',
+			'type'              => 'string',
+			'items'             => array(
+				'enum' => array( 'unread', 'dismissed' ),
+				'type' => 'string',
+			),
+			'sanitize_callback' => 'sanitize_key',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		$params['type'] = array(
+			'description'       => __( 'Limit result set to items with a specific type.', 'buddypress' ),
+			'default'           => 'active',
+			'type'              => 'string',
+			'items'             => array(
+				'enum' => array( 'active', 'inactive' ),
+				'type' => 'string',
+			),
+			'sanitize_callback' => 'sanitize_key',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		$params['target'] = array(
+			'description'       => __( 'Limit result set to items concerning one or more specific targets.', 'buddypress' ),
+			'type'              => 'array',
+			'items'             => array(
+				'enum' => array( 'community', 'contributors', 'admins' ),
+				'type' => 'string',
+			),
+			'sanitize_callback' => 'wp_parse_list',
+			'validate_callback' => 'rest_validate_request_arg',
+		);
+
+		$params['priority'] = array(
+			'description'       => __( 'Limit result set to items having a specific priority.', 'buddypress' ),
+			'type'              => 'integer',
+			'items'             => array(
+				'enum' => array( 1, 2, 3 ),
```

Comment:
So it is not possible to fetch other types of items as an admin?

Reply:
Absolutely 😄